src/reduce.js

```diff
@@ -6,6 +6,40 @@ import {abs, exp, log} from './unary.js';
 import {sizeOfShape, Scalar, Tensor} from './lib/tensor.js';
 import {validateReduceParams} from './lib/validate-input.js';
 
+export function selectValuesToReduce(input, axes, inputLocation) {
+  validateReduceParams(input, {axes});
+
+  const outputShape = input.shape.slice();
+  for (let i = 0; i < axes.length; ++i) {
+    outputShape[axes[i]] = 1;
+  }
+
+  // Calculate the "strides" across the reduction dimensions given in axes.
+  axes.sort((a, b) => a - b);
+  const reduceDims = axes.map((axis) => input.shape[axis]);
+  const reduceElements = sizeOfShape(reduceDims);
+  const reduceStrides = new Array(axes.length);
+  reduceStrides[reduceStrides.length - 1] = 1;
```

Comment:
Appears this will fail on the `axes = []` case, which should return exactly one element of shape `[]`. *Most* of the logic already looks correct to handle that case (including `sizeOfShape` which will return 1).

Reply:
It's strange. It can work for this case without changes, my understanding is that index should be start from **0**.
```js
reduceStrides = [];
reduceStrides[-1] = 1; // it can work
```

But I add `if` condition block before this, then it never goes here for **axes=[]**, please see the update commit, thanks.

Reply:
> It's strange. It can work for this case without changes
> reduceStrides[-1] = 1; // it can work

Huh, what a weird language: https://stackoverflow.com/questions/58570277/adding-an-element-in-negative-index-of-an-array

Reply:
`reduceStrides[-1] = 1` getting `[-1: 1]` is bad, we should avoid that.

Reply:
Thanks @huningxin!
I've updated commit to avoid it, please take another look, thanks.